python/test/gcs/test_gcs_to_jdbc.py

```diff
@@ -0,0 +1,157 @@
+"""
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+"""
+
+import mock
+import pyspark
+
+from dataproc_templates.gcs.gcs_to_jdbc import GCSToJDBCTemplate
+import dataproc_templates.util.template_constants as constants
+
+
+class TestGCSToBigQueryTemplate:
+    """
+    Test suite for GCSToJDBCTemplate
+    """
+
+    def test_parse_args(self):
+        """Tests GCSToJDBCTemplate.parse_args()"""
+
+        gcs_to_jdbc_template = GCSToJDBCTemplate()
+        parsed_args = gcs_to_jdbc_template.parse_args(
+            ["--gcs.jdbc.input.format=parquet",
+             "--gcs.jdbc.input.location=gs://test",
+             "--gcs.jdbc.output.table=table",
+             "--gcs.jdbc.output.mode=append",
+             "--gcs.jdbc.output.driver=driver",
+             "--gcs.jdbc.output.url=url"])
+
+        assert parsed_args["gcs.jdbc.input.format"] == "parquet"
+        assert parsed_args["gcs.jdbc.input.location"] == "gs://test"
+        assert parsed_args["gcs.jdbc.output.table"] == "table"
+        assert parsed_args["gcs.jdbc.output.mode"] == "append"
+        assert parsed_args["gcs.jdbc.output.driver"] == "driver"
+        assert parsed_args["gcs.jdbc.output.url"] == "url"
+
+    @mock.patch.object(pyspark.sql, 'SparkSession')
+    def test_run_parquet(self, mock_spark_session):
+        """Tests GCSToJDBCTemplate runs with parquet format"""
+
+        gcs_to_jdbc_template = GCSToJDBCTemplate()
+        mock_parsed_args = gcs_to_jdbc_template.parse_args(
+            ["--gcs.jdbc.input.format=parquet",
+             "--gcs.jdbc.input.location=gs://test",
+             "--gcs.jdbc.output.table=table",
+             "--gcs.jdbc.output.mode=append",
+             "--gcs.jdbc.output.driver=driver",
+             "--gcs.jdbc.output.url=url"])
+        mock_spark_session.read.parquet.return_value = mock_spark_session.dataframe.DataFrame
+        gcs_to_jdbc_template.run(mock_spark_session, mock_parsed_args)
+
+        mock_spark_session.read.parquet.assert_called_once_with("gs://test")
+        mock_spark_session.dataframe.DataFrame.write.format.assert_called_once_with(
+            constants.FORMAT_JDBC)
+        mock_spark_session.dataframe.DataFrame.write.format(
```

Comment:
These tests do not match the .write implementation:
For example, here you want to test the "dbtable" option, but if the test is ...DataFrame.write.format().option., you are testing the first option after the format(), which is "url"
Also, the constants.TABLE is "table", right? And not "dbtable" as required.
Follow the order of the implementation:
input_data.write \
            .format(constants.FORMAT_JDBC) \
            .option("url", jdbc_url) \
            .option("dbtable", jdbc_table) \
            .option("driver", output_driver) \
            .option("batchsize", batch_size) \
            .mode(output_mode) \
            .save()

Reply:
Done